test/hotspot/jtreg/serviceability/jvmti/HeapMonitor/MyPackage/HeapMonitor.java

```diff
@@ -164,7 +164,21 @@ public static void freeStorage() {
     enableSamplingEvents();
     setSamplingInterval(0);
 
-    // Loop around an allocation loop and wait until the tlabs have settled.
+    // Loop around an allocation loop and wait until the TLABs have settled.
+    // It takes two steps to enable Object Sampling with interval 0
+    // 1. consume current TLAB, which can be varies with heap/GC configuration
+    //    then allocation enters allocate_inside_tlab_slow and check
+    //    JvmtiExport::should_post_sampled_object_alloc
+    // 2. consume initial ThreadHeapSampler::_bytes_until_sample
+    //    then runtime invokes pick_next_sample and check sampleing interval
+    //    and set _bytes_until_sample to 0.
+    //    initial _bytes_until_sample is geometric variable with the specified mean
+    //    (512K by default), check ThreadHeapSampler::pick_next_geometric_sample()
+    //
+    // trigger GC to consume current TLAB in step1
+    // consume initial _bytes_until_sample in following loops, each iteration consume
+    // about 1600KB, 10 iterations can definitly consume intitial _bytes_until_sample
```

Comment:
I think this comment should be after the System.gc() call.

Reply:
Thanks and accepted.